classes/user_extractor.php

```diff
@@ -76,7 +76,23 @@ public static function get_user(string $fieldname, string $fieldvalue, bool $ins
             // Check if requested field exists, required for Totara compatibility.
             $fields = array_merge(\core_user::AUTHSYNCFIELDS, ['id', 'username']);
             if (in_array($fieldname, $fields)) {
-                $user = get_complete_user_data($fieldname, $fieldvalue);
+
+                $fieldsql = " AND ". $DB->sql_equal($fieldname, ':fieldvalue', !$insensitive);
+                $params['fieldname'] = $fieldname;
+                $params['fieldvalue'] = $fieldvalue;
+                $params['mnethostid'] = $CFG->mnet_localhost_id;
+
+                $sql = "SELECT id
+                            FROM {user}
+                            WHERE deleted <> 1 AND
+                                mnethostid = :mnethostid $fieldsql";
+
+                if ($records = $DB->get_records_sql($sql, $params)) {
```

Comment:
Perhaps you can use `$DB->get_records_sql($sql, $params, IGNORE_MULTIPLE)` here instead, with the `IGNORE_MULTIPLE` flag to only use the first result if one exists. This should simplify the logic in this block.

Reply:
But if more than  one record we exit right? And don't allow the match? Your example would allow a match when more than one user is possible to select from?

Reply:
Ah, you're right. I missed the intention. The alternative here would be using the `MUST_EXIST` flag, however that will look like making a code flow work with exception handling which is even uglier.

In this case, the only suggest would be the use `===` instead of `==` for the count check.